### ISSUE TITLE
Duplicate state name is not caught during validation

### ISSUE THREAD

**Issue by massfords** [ChristopheBougere/asl-validator#80]
**Describe the bug**
State names must be unique within the scope of the whole state machine. 

**To Reproduce**
PR is inbound w/ a test
```json
{
  "Comment": "Exhibits duplicate state error. State names MUST be unique within the scope of the whole state machine.",
  "StartAt": "Map",
  "States": {
    "Map": {
      "Type": "Map",
      "Next": "Final State",
      "InputPath": "$.input",
      "ItemsPath": "$.items",
      "MaxConcurrency": 0,
      "Iterator": {
        "StartAt": "Final State",
        "States": {
          "Final State": {
            "Type": "Wait",
            "Seconds": 20,
            "End": true
          }
        }
      },
      "ResultPath": "$.result"
    },
    "Final State": {
      "Type": "Pass",
      "End": true
    }
  }
}
```

**Expected behavior**
Report an error

**Version:** 
2.0.0

**Additional context**

**Comment by massfords** [comment]
I'd like to include a minor refactor in this issue to make it easier to add new rules.

The small change is in the validator.js to move all of the static analysis checks to after schema validation and track all of their errors in a single array. 

See the branch here: https://github.com/massfords/asl-validator/blob/unique-state-names/src/validator.js

Here's the gist of it:

```
  const postSchemaValidationErrors = [];
  if (isJsonSchemaValid) {
    postSchemaValidationErrors.push(...checkJsonPath(definition));
    postSchemaValidationErrors.push(...missingTransitionTarget(definition));
    postSchemaValidationErrors.push(...stateTransitions(definition));
    postSchemaValidationErrors.push(...stateNames(definition));       // <--- new rule for this ticket
  }
```

**Comment by ChristopheBougere** [comment]
Great idea, these checks were added with time and it became a bit messy.

Your change looks good to me, can you open a PR?
Just a question: I see you're doing additional checks only if the JSON schema is valid. Is there a specific reason for that? Is this AWS SFN behavior? Or it's because they might fail and give false positives?

**Comment by massfords** [comment]
Deferring the checks is mainly to avoid errors in the checkers due to missing or invalid values. It's easier to write the checkers knowing the document is schema valid. 

I'll open a PR shortly for this. I may have some other rules to include.

**Comment by ChristopheBougere** [comment]
Released in asl-validator@2.1.0
